lineapy/graph_reader/session_artifacts.py

```diff
@@ -413,7 +433,71 @@ def _slice_session_artifacts(self) -> None:
         )
         self.input_parameters_nodecollection._update_graph(self.graph)
 
+    def _update_nodecollection_dependencies(self):
+        """
+        Update nodecollection dependencies when we replace the cached artifact
+        nodecollection with `lineapy.get`.
+        """
+        last_appearrance_nc: Dict[str, str] = dict()
+        dependencies: Dict[str, Set[str]] = dict()
+        # Artifact nodes that are going to be replaced by cached value
+        cache_nodes = [
+            nc.name
+            for nc in self.artifact_nodecollections
+            if nc.use_cache is not None
+        ]
+        # Determine input variables of a nodecollection are coming from output
+        # variables of nodecollections
+        for nc in self.artifact_nodecollections:
+            dependencies[nc.name] = set()
+            for var in nc.input_variables:
+                if var in last_appearrance_nc.keys():
+                    dependencies[nc.name].add(last_appearrance_nc[var])
+            for var in nc.return_variables:
+                last_appearrance_nc[var] = nc.name
+        # Nodecollection dependencies
```

Comment:
Remove this comment or change it. 
Maybe something like 

Create nodecollection_dependencies taskgraph from dependencies